docs/grammatic.txt

```diff
@@ -0,0 +1,225 @@
+=====================================
+    This is grammatic of MySwift:
+=====================================
+
+==================
+    TODO LIST:
+
+==================
+
+<program> ::=
+    <global-list>
+
+<global-list> ::=
+    <class-declaration> [ <global-list> ] |
+    <function-declaration> [ <global-list> ] |
+    <variable-declaration> [ <global-list> ] |
+    <constant-declaration> [ <global-list> ]
+
+<class-declaration> ::=
+    "class" <class-name> [ <type-inheritance-clause> ] <class-body>
+
+<type-inheritance-clause> ::=
+    ":" <type-inheritance-list>
+
+<type-inheritance-list> ::=
+    <type-identifier> |
+    <type-identifier> , <type-inheritance-list>
+
+<class-name> ::=
+    <identifier>
+
+<class-body> ::=
+    "{" <class-members> "}"
+
+<class-members> ::=
+    <class-member> [ <class-members> ]
+
+<class-member> ::=
+    <declaration>
+
+<extension-declaration> ::=
+    "extension" <type-identifier> [ <type-inheritance-clause> ] <extension-body>
+
+<extension-body> ::=
+    "{" <extension-members> "}"
+
+<extension-members> ::=
+    <extension-member> [ <extension-members> ]
+
+<extension-member> ::=
+    <declaration>
+
+<expression-list> ::=
+    <expression> |
+    <expression> "," <expression-list>
+
+<type> ::=
+    <type-identifier> |
+    "[" <type-identifier> "]"
+
+<type-identifier> ::=
+    <type-name> |
+    <type-name> "." <type-identifier>
```

Comment:
What does it mean? Is it for nested classes?

Reply:
Yes, it is for nested classes or types
For example:
class A {
 class B;
}

A.B